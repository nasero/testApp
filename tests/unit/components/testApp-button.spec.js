import { shallowMount } from '@vue/test-utils';
import TestAppButton from '@/components/testApp-button';

describe('TestAppButton - Snapshot', () => {
    it('Snapshot', () => {
        const text = 'my button';
        const wrapper = shallowMount(TestAppButton, {
            propsData: { text }
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});

describe('TestAppButton - methods', () => {
    it('btnClicked', async () => {
        const text = 'button';
        const wrapper = shallowMount(TestAppButton, {
            propsData: { text }
        });

        wrapper.vm.buttonClicked();

        await wrapper.vm.$nextTick();

        expect(wrapper.emitted('testApp-button-clicked')).toBeTruthy();
    });
});
