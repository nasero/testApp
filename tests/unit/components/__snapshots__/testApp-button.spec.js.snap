// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestAppButton - Snapshot Snapshot 1`] = `
<button
  class="testApp-button"
  type="button"
>
  my button
</button>
`;
